### PR TITLE
feat(client): add connect_or_create

### PR DESCRIPTION
**Pull request by AdeelK93**
## Change Summary

Adds connectOrCreate to support #754

Please let me know if I should make write tests (and where to put them). Thanks @RobertCraigie for your work on this great library!

## Checklist

- [x] Unit tests for the changes exist
- [ ] Tests pass without significant drop in coverage
- [ ] Documentation reflects changes where applicable
- [ ] Test snapshots have been [updated](https://prisma-client-py.readthedocs.io/en/latest/contributing/contributing/#snapshot-tests) if applicable

## Agreement

By submitting this pull request, I confirm that you can use, modify, copy and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/prisma/generator/templates/types.py.jinja [hunk 1]
```diff
@@ -479,16 +479,20 @@ class {{ model.name }}CreateWithoutRelationsInput({{ model.name }}OptionalCreate
...
     {% endfor %}
 
+class {{ model.name }}ConnectOrCreateWithoutRelationsInput(TypedDict):
+    create: '{{ model.name }}CreateWithoutRelationsInput'
+    where: '{{ model.name }}WhereUniqueInput'
 
 class {{ model.name }}CreateNestedWithoutRelationsInput(TypedDict, total=False):
     create: '{{ model.name }}CreateWithoutRelationsInput'
     connect: '{{ model.name }}WhereUniqueInput'
+    connectOrCreate: '{{ model.name }}ConnectOrCreateWithoutRelationsInput'
```
Comment (RobertCraigie) [inline src/prisma/generator/templates/types.py.jinja:489]: nit: I think this should be snake case, `connect_or_create`. You'll need to update the `QUERY_BUILDER_ALIASES` mapping in `src/prisma/_constants.py` to rename it back to `connectOrCreate` internally

**Review by RobertCraigie — COMMENTED** [review]
Thanks @AdeelK93!

r.e. testing, you should add tests to `databases/tests/test_create.py` and `databases/sync_tests/test_create.py`. See [these docs](https://prisma-client-py.readthedocs.io/en/stable/contributing/contributing/#database-tests) for information on the database testing setup and let me know if you need any help, I don't think all the setup there is particularly well documented right now.

**Comment by AdeelK93** [comment]
thanks @RobertCraigie - I moved the tests to the files you mentioned, and renamed the function to snake case. Any other feedback?

**Comment by AdeelK93** [comment]
hi @RobertCraigie, looks like one test failed, but it’s a transient error (incomplete read while downloading node binary). the tests all passed on my fork, could we please retry?

**Review by RobertCraigie — APPROVED** [review]
Awesome, thanks! Great work :)

**Pull request merged**